Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CorrelationAttributeUtil.java

```diff
@@ -158,6 +192,51 @@ private static String getEmailAddressAttrDisplayName() {
         return correlationAttrs;
     }
 
+    /**
+     * Makes a correlation attribute instance from a phone number attribute of an
+     * artifact.
+     *
+     * @param artifact An artifact with a phone number attribute.
+     *
+     * @return The correlation instance artifact or null, if the phone number is
+     *         not a valid correlation attribute.
+     *
+     * @throws TskCoreException     If there is an error querying the case
+     *                              database.
+     * @throws CentralRepoException If there is an error querying the central
+     *                              repository.
+     */
+    private static void makeCorrAttrFromArtifactPhoneAttr(List<CorrelationAttributeInstance> corrAttrInstances, BlackboardArtifact artifact) throws TskCoreException, CentralRepoException {
+        CorrelationAttributeInstance corrAttr = null;
+
+        /*
+         * Extract the phone number from the artifact attribute.
+         */
+        String value = null;
+        if (null != artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER))) {
+            value = artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER)).getValueString();
+        } else if (null != artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_FROM))) {
+            value = artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_FROM)).getValueString();
+        } else if (null != artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_TO))) {
+            value = artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_PHONE_NUMBER_TO)).getValueString();
+        }
```

Comment:
I realize that the code above is restored code, but it looks like it contains another bug. The contact book entry, call log entry and message artifacts all support multiple phone number attributes. I think we need to loop over all of the attributes in the artifact, rather than stop after "harvesting" one phone number.

Reply:
See Raman's comment for new story proposal.

Reply:
@dannysmyda, please create the story or stories, assign to self, etc. Thanks!

Reply:
I'll just do it here then.